system/jlib/jsocket.cpp

```diff
@@ -6428,7 +6442,15 @@ class CSocketConnectWait: implements ISocketConnectWait, public CInterface
                 else
                 {   // select/poll failed
                     err = ERRNO();
-                    LOGERR(err,2,"CSocketConnectWait ::select/poll");
+                    if (err != JSE_INTR)
+                    {
+                        LOGERR(err,2,"CSocketConnectWait ::select/poll");
+                        sock->errclose();
+                        isopen = false;
+                        if (!oneshot)
+                            connectimedout = true; // to force same behavior as before
```

Comment:
Before it would loop until timed out and then connectimedout would be true.  If we break out here early then set it to true to have same return behaviour.

Reply:
I could understand the change I think if it was:
```
    if (oneshot) 
        connecttimedout = true;
```

but with oneshot == false, it's supposed to loop and retry pre_connect afaics.